libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Maxell.h

```diff
@@ -15,6 +15,10 @@ class AP_BattMonitor_SMBus_Maxell : public AP_BattMonitor_SMBus
                              AP_BattMonitor::BattMonitor_State &mon_state,
                              AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev);
 
+    // generic SMBus batteries may not always report the cell voltages, as we
+    // the data is being read from manufacturer specific fields
```

Comment:
I was the first person saying that I desired a common SMBus driver and I still think that can be achieved by determining in run-time what battery it is.
But I don't understand the difference you make between the Solo and Maxell drivers. You say that Maxell uses one method that isn't in the standard spec and Solo uses two, how does that make them different? One method is the difference between being a generic driver and being useless to any other chip maker? The only thing strange in Solo spec is the current register not being the standard, but other than that it is as generic as Maxell (that, in my opinion, isn't).

I think where we disagree is that you are seeing the Solo battery as the only battery that will ever use those two methods, while the Maxell one will be used in more. But I don't see it that way, you might have not found the same chip that is in Solo batteries in other ones, but it doesn't mean it doesn't or won't exist.
Also, as this market starts to expand, we will probably see more chips and other non-standard methods will appear.